Junit sw

sw801733 opened the pull request:
feat : 암호 강도 측정 기능 추가

sw801733 committed:
feat : 암호 강도 측정 기능 추가

- 비밀번호의 길이로 여덟 글자 이상 검증
- 정규 표현식으로 0~9 사이의 숫자 포함 여부 검증
- 정규 표현식으로 대문자 포함 여부 검증

hjp1016 commented:
코드가 기똥차네여 저도 알려주세요

NotableDeveloper commented on src/main/java/PasswordStrength/PasswordValidator.java:
```diff
@@ -16,6 +16,39 @@ public void setPassword(String password) {
+      caseCnt++;
+    }
+
+    if (caseCnt == 1)
```
다중 if문을 아래처럼 간결하게 바꾸면 좋을 것 같습니다.

if() return ...
if() return ...
retrun ...

NotableDeveloper commented on src/main/java/PasswordStrength/PasswordValidator.java:
```diff
@@ -16,6 +16,39 @@ public void setPassword(String password) {
+    }
+
+    // 암호는 0부터 9사이의 숫자를 포함한다.
+    if (password.matches(".*[0-9].*")){
```
별도의 private method로 작성하면 좋을 것 같습니다.

NotableDeveloper commented on src/main/java/PasswordStrength/PasswordValidator.java:
```diff
@@ -16,6 +16,39 @@ public void setPassword(String password) {
+    }
+
+    // 암호는 대문자를 포함한다.
+    if (password.matches(".*[A-Z].*")){
```
별도의 private method로 작성하면 좋을 것 같습니다.

NotableDeveloper commented on src/main/java/PasswordStrength/PasswordValidator.java:
```diff
@@ -16,6 +16,39 @@ public void setPassword(String password) {
+    // 암호는 여덟 글자 이상이어야 한다.
+    int passwordLength = password.length();
+
+    if (passwordLength >= 8) {
```
별도의 private method로 작성하면 좋을 것 같습니다.

sw801733 replied:
별도 메서드 고민했었는데 바로 수정해 보겠습니다!

sw801733 committed:
refact : 암호 검증 로직 메서드로 추출

- 글자 개수 검증 checkLength 메서드로 추출
- 0~9 숫자 포함 검증 containsDigit 메서드로 추출
- 대문자 포함 검증 containsUpper 메시드로 추출

NotableDeveloper commented on src/main/java/PasswordStrength/PasswordValidator.java:
```diff
@@ -15,7 +15,38 @@ public void setPassword(String password) {
+    // 암호는 대문자를 포함한다.
+    if (containsUpper()) caseCnt++;
+
+    if (caseCnt == 1) return PasswordStrength.WEAK;
```
아래와 같이 caseCnt를 2부터 검사하면 한 줄의 if문을 줄일 수 있을 것 같습니다.

 if (caseCnt == 2) return PasswordStrength.NORMAL;
 if (caseCnt == 3) return PasswordStrength.STRONG;
 else return PasswordStrength.WEAK;

sw801733 committed:
refact : 암호 검증 케이스 조건문 수정

- caseCnt 의 비교를 2부터 시작해서 if 문 조건 1개 생략 가능
